src/drivers/intel/hda/hda-dma.c

```diff
@@ -309,9 +296,8 @@ static void hda_dma_post_copy(struct dma_chan_data *chan, int bytes)
 		 */
 		hda_dma_inc_fp(chan, bytes);
 
-		/* Force Host DMA to exit L1 if scheduled on DMA */
-		if (!hda_chan->irq_disabled)
-			pm_runtime_put(PM_RUNTIME_HOST_DMA_L1, 0);
+		/* Force Host DMA to exit L1 */
+		pm_runtime_put(PM_RUNTIME_HOST_DMA_L1, 0);
```

Comment:
I must admit I am quite confused...
I can see that for the LMEM-HMEM transfer you can do the L1 exit after copying the data, e.g. to force data to be transfered. 

however, if you do a HMEM-LMEM transfer, does it really make sense to exit L1 *after* copying the data?